### PR TITLE
Make markerfacecolor work for 3d scatterplots  

### PR DESCRIPTION
**Pull request by jinshifen33**
## PR Summary
  This pull request is a fix for [issue 9725](https://github.com/matplotlib/matplotlib/issues/9725) color issue on 3D scatter plots. Invoking the set color method after the initial creation of a scatter 3D plot would not change the color; these changes fix this issue.
 
## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is PEP 8 compliant
- [x] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/mpl_toolkits/mplot3d/art3d.py [hunk 1]
```diff
@@ -449,24 +450,62 @@ def set_3d_properties(self, zs, zdir):
         self._edgecolor3d = self.get_edgecolor()
         self.stale = True
 
+    def set_facecolor(self, colors):
+        # override setter for facecolor so that 3dfacecolor
+        # will be updated as well. 
+        # In providing consitentcy in 2dcolor and 3dcolor setter.
+        # Note: will also update edgecolor if it follows facecolor
+
+        PathCollection.set_facecolor(self, colors)
```
Comment (timhoffm) [inline lib/mpl_toolkits/mplot3d/art3d.py]: Since this will by py3 only, use super().

**Comment by jinshifen33** [comment]
One thing I noticed is that, the edge_color will never equal to  "face" in 3D, this fix also introduces a workaround for it, I am not sure whether this is desired behaviour .

File: lib/mpl_toolkits/mplot3d/art3d.py [hunk 2]
```diff
@@ -449,24 +450,62 @@ def set_3d_properties(self, zs, zdir):
...
+    def set_edgecolor(self, colors):
+        # override setter for edgecolor so that 3dedgecolor
+        # will be updated as well.
+        # In providing consitentcy in 2dcolor and 3dcolor setter.
+
+        PathCollection.set_edgecolor(self, colors)
+        self._edgecolor3d = PathCollection.get_edgecolor(self)
+    set_edgecolors = set_edgecolor
+
+    def _set_facecolors_on_project(self, colors):
```
Comment (timhoffm) [inline lib/mpl_toolkits/mplot3d/art3d.py]: what is `project`? Should this rather be `_set_facecolors_on_projection` or `_set_facecolors_on_2d_projection`?

File: lib/mpl_toolkits/mplot3d/art3d.py [hunk 3]
```diff
@@ -449,24 +450,62 @@ def set_3d_properties(self, zs, zdir):
...
+        # A sepearted method that changes edgecolor in drawing
```
Comment (timhoffm) [inline lib/mpl_toolkits/mplot3d/art3d.py]: facecolor

File: lib/mpl_toolkits/mplot3d/art3d.py [hunk 4]
```diff
@@ -449,24 +450,62 @@ def set_3d_properties(self, zs, zdir):
...
+
     def do_3d_projection(self, renderer):
         xs, ys, zs = self._offsets3d
         vxs, vys, vzs, vis = proj3d.proj_transform_clip(xs, ys, zs, renderer.M)
 
         fcs = (zalpha(self._facecolor3d, vzs) if self._depthshade else
                self._facecolor3d)
         fcs = mcolors.to_rgba_array(fcs, self._alpha)
-        self.set_facecolors(fcs)
+        # now uses setter on project
```
Comment (timhoffm) [inline lib/mpl_toolkits/mplot3d/art3d.py]: The comment does not add information and thus can be left out.

**Comment by jklymak** [comment]
Milestoning 3.0 because this isn't fixing a regression.

**Comment by jklymak** [comment]
Thanks a lot for your contribution.

However, this conflicts with an existing PR:  #10489  I'm going to close this PR to reduce confusion.  If you want to help, you can make comments on #10489.  (Also if you want to see if someone else has made a PR based on an issue, its usually cross-linked at the bottom of the issue.)

**Pull request closed without merging**